tests/jobs/test_scheduler_job.py

```diff
@@ -1274,7 +1274,8 @@ def test_dagrun_timeout_fails_run_and_update_next_dagrun(self, dag_maker):
         session.flush()
         session.refresh(dr)
         assert dr.state == State.FAILED
-        # check that next_dagrun has been updated by Schedulerjob._update_dag_next_dagruns
+        self.scheduler_job._do_scheduling(session)
```

Comment:
I'm not sure if this test is correct or not, the assert below was failing, that was why I made the change here. Could be another bug, please review